synapse/handlers/saml_handler.py

```diff
@@ -304,6 +302,35 @@ async def grandfather_existing_users() -> Optional[str]:
                 grandfather_existing_users,
             )
 
+    def _remote_id_from_saml_response(
+        self,
+        saml2_auth: saml2.response.AuthnResponse,
+        client_redirect_url: Optional[str],
+    ) -> str:
+        """Extract the unique remote id from a SAML2 AuthnResponse
+
+        Args:
+            saml2_auth: The parsed SAML2 response.
+            client_redirect_url: The redirect URL passed in by the client.
+        Returns:
+            remote user id
+
+        Raises:
+            MappingException if there was an error extracting the user id
+        """
+        # It's not obvious why we need to pass in the redirect URI to the mapping
+        # provider, but we do :/
```

Comment:
Do you see any use for it? I don't think any of the custom mapping providers I know of use it. I guess there isn't a huge harm in keeping it though.

Reply:
No, I don't see any use for it, but changing the api for the mapping providers is annoying.